src/Fixer/Whitespace/NoTrailingWhitespaceFixer.php

```diff
@@ -62,7 +62,25 @@ public function isCandidate(Tokens $tokens)
      */
     protected function applyFix(\SplFileInfo $file, Tokens $tokens)
     {
-        foreach ($tokens as $index => $token) {
+        for ($index = count($tokens) - 1; $index >= 0; --$index) {
+            $token = $tokens[$index];
+            if (
+                $token->isGivenKind(T_OPEN_TAG)
+                && $tokens->offsetExists($index + 1)
+                && $tokens[$index + 1]->isWhitespace()
+                && 1 === Preg::match('/(.*)\h$/', $token->getContent(), $openTagMatches)
+                && 1 === Preg::match('/^(\R)(.*)$/', $tokens[$index + 1]->getContent(), $whitespaceMatches)
+            ) {
+                $tokens[$index] = new Token([T_OPEN_TAG, $openTagMatches[1].$whitespaceMatches[1]]);
+                if (empty($restOfWhitespace)) {
```

Comment:
can we avoid `empty` here? , `$restOfWhitespace` is not defined here I think?

Reply:
Fixed